### PR TITLE
Update native AOT index.md for .NET 9

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
## Summary

* Drop .NET 7
* Add .NET 9

Cc @dotnet/ilc-contrib 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/deploying/native-aot/index.md](https://github.com/dotnet/docs/blob/45b7ac8677687e7b9c4a7b8ee841be70706f863c/docs/core/deploying/native-aot/index.md) | [docs/core/deploying/native-aot/index](https://review.learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/index?branch=pr-en-us-42547) |

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
Technically we could also update the prerequisites section since we no longer need zlib-dev to build but not sure if it's worth adding complexity as long as .NET 8 is supported.

**Review by jkotas — APPROVED** [review]

File: docs/core/deploying/native-aot/index.md [hunk 1]
```diff
@@ -14,16 +14,6 @@ The benefit of Native AOT is most significant for workloads with a high number o
...
-
-### [.NET 7](#tab/net7)
-
-Native AOT is targeted towards console-type apps. Only a limited number of libraries are fully compatible with Native AOT.
-
-### [.NET 8+](#tab/net8plus)
-
-AOT support in .NET 8 is more comprehensive than in .NET 7. However, there are still some limitations. For more information, see [Limitations of Native AOT deployment](#limitations-of-native-aot-deployment).
-
 ---
```
Comment (MichalStrehovsky) [inline docs/core/deploying/native-aot/index.md:17]:
```suggestion
```

File: docs/core/deploying/native-aot/index.md [hunk 2]
```diff
@@ -154,35 +120,16 @@ Native AOT apps have the following limitations:
...
-
-### [.NET 7](#tab/net7)
-
-- Should be targeted for console type apps. ASP.NET Core is ***not*** supported.
-- Limited diagnostic support for debugging and profiling.
-
-### [.NET 8+](#tab/net8plus)
-
-- [Diagnostic support for debugging and profiling](./diagnostics.md) with some limitations.
+- - [Diagnostic support for debugging and profiling](./diagnostics.md) with some limitations.
```
Comment (MichalStrehovsky) [inline docs/core/deploying/native-aot/index.md:123]:
```suggestion
- [Diagnostic support for debugging and profiling](./diagnostics.md) with some limitations.
```

File: docs/core/deploying/native-aot/index.md [hunk 3]
```diff
@@ -196,4 +141,18 @@ The following table shows supported compilation targets.
...
+| Platform | Supported architecture |  Notes               |
+|----------|------------------------|----------------------|
+| Windows  | x64, Arm64, x86        |                      |
+| Linux    | x64, Arm64, Arm        |                      |
+| macOS   | x64, Arm64 |                                        |
+| iOS     | Arm64 | Experimental support                   |
+| iOSSimulator     | x64, Arm64 | Experimental support                   |
+| tvOS     | Arm64 | Experimental support                   |
+| tvOSSimulator     | x64, Arm64 | Experimental support                   |
+| MacCatalyst     | x64, Arm64 | Experimental support                   |
```
Comment (ivanpovazan) [inline docs/core/deploying/native-aot/index.md:155]:
```suggestion
| iOS     | Arm64 |                   |
| iOSSimulator     | x64, Arm64 |                   |
| tvOS     | Arm64 |                    |
| tvOSSimulator     | x64, Arm64 |                    |
| MacCatalyst     | x64, Arm64 |                   |
```

I think this comment should be removed, as all the work on supporting NativeAOT with MAUI apps targeting iOS platforms has been completed and we will officially support it in .NET 9 release.

File: docs/core/deploying/native-aot/index.md [hunk 4]
```diff
@@ -154,35 +118,16 @@ Native AOT apps have the following limitations:
...
-- Should be targeted for console type apps. ASP.NET Core is ***not*** supported.
-- Limited diagnostic support for debugging and profiling.
-
-### [.NET 8+](#tab/net8plus)
-
 - [Diagnostic support for debugging and profiling](./diagnostics.md) with some limitations.
 - Support for some ASP.NET Core features. For more information, see [ASP.NET Core support for Native AOT](/aspnet/core/fundamentals/native-aot/).
 
----
+The publish process analyzes the entire project and its dependencies for possible limitations. Warnings are issued for each limitation the published app may encounter at run time.
```
Comment (gewarren) [inline docs/core/deploying/native-aot/index.md:124]:
```suggestion
The publish process analyzes the entire project and its dependencies for possible limitations. Warnings are issued for each limitation the published app might encounter at run time.
```

**Review by gewarren — APPROVED** [review]
Thanks!

**Pull request merged**